tests/blobcache.bats

```diff
@@ -55,6 +55,29 @@ function _check_matches() {
         expect_output --from="$unmatched" "$5"  "$6 should not match"
 }
 
+# Integration test for https://github.com/containers/image/pull/1645
+@test "blobcache: blobs must be reused when pushing across registry" {
+	start_registry
+	run_buildah login --tls-verify=false --authfile ${TEST_SCRATCH_DIR}/test.auth --username testuser --password testpassword localhost:${REGISTRY_PORT}
+	run_buildah pull registry.fedoraproject.org/fedora-minimal
+	run_buildah tag registry.fedoraproject.org/fedora-minimal localhost:${REGISTRY_PORT}/test
+	run_buildah --log-level debug push --tls-verify=false --authfile ${TEST_SCRATCH_DIR}/test.auth localhost:${REGISTRY_PORT}/test docker://localhost:${REGISTRY_PORT}/test
+	# must not contain "Skipping blob" since push must happen
+	assert "$output" !~ "Skipping blob"
+	run_buildah --log-level debug push --tls-verify=false --authfile ${TEST_SCRATCH_DIR}/test.auth localhost:${REGISTRY_PORT}/test docker://localhost:${REGISTRY_PORT}/test
+	expect_output --substring "Skipping blob"
+
+	# Clear local image and c/image's blob-info-cache
+	run_buildah rmi --all -f
+	run rm /var/lib/containers/cache/blob-info-cache-v1.sqlite
```

Comment:
The result of this run() isn't checked, and I'd be surprised if it succeeds for non-root users.

Reply:
@nalind New commit addresses this comment.